docs/docsite/Makefile

```diff
@@ -50,23 +59,47 @@ all: docs
 
 docs: htmldocs
 
+coredocs: core_htmldocs
+
 generate_rst: collections_meta config cli keywords plugins testing
-base_generate_rst: collections_meta config cli keywords base_plugins testing
+core_generate_rst: collections_meta config cli keywords base_plugins testing
+
+# The following two symlinks are necessary to produce two different docsets
+# from the same set of rst files (Ansible the package docs, and core docs).
+# Symlink the relevant index into place for building Ansible docs
+ansible_structure: generate_rst
+	# We must have python and python-packaging for the version_helper
+	# script so use it for version comparison
+	if python -c "import sys, packaging.version as p; sys.exit(not p.Version('$(MAJOR_VERSION)') >  p.Version('2.10'))" ; then \
+		echo "Creating symlinks in generate_rst"; \
+		ln -sf ../rst/ansible_index.rst rst/index.rst; \
+		ln -sf ../sphinx_conf/ansible_conf.py rst/conf.py; \
+	else \
+		echo 'Creating symlinks for older ansible in generate_rst'; \
+		ln -sf ../rst/2.10_index.rst rst/index.rst; \
+		ln -sf ../sphinx_conf/2.10_conf.py rst/conf.py; \
+	fi
+
+# Symlink the relevant index into place for building core docs
+core_structure: core_generate_rst
+	@echo "Creating symlinks in core_generate_rst"
+	-ln -sf ../rst/core_index.rst rst/index.rst
+	-ln -sf ../sphinx_conf/core_conf.py rst/conf.py
```

Comment:
I don't think symlinking is necessary. Just pass different configs and settings via CLI args.
Look:
```console
$ sphinx-build --help
usage: sphinx-build [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]

[...]


build configuration options:
  -c PATH           path where configuration file (conf.py) is located
                    (default: same as SOURCEDIR)
[...]
  -D setting=value  override a setting in configuration file
[...]
```

So you could invoke something like
```console
$ python -m sphinx -c sphinx_conf/2.10_conf.py -D master_doc=ansible_index ...
```
and there would be no need to mutate files on disk.

Reply:
Also, we could explore if using tags (https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#tags) for conditional ToCs...

Reply:
fwiw - both were tried and rejected by others in favor of the symlink approach. We'll see what other comments there are.

Reply:
These symlinks appear to be responsible for the package-data sanity test failure. Instead of linking them, have you tried copying them instead?

Reply:
Tried the copy instead of symlink localy and that does seem to work and removes the build errors, thanks!

Reply:
Hmmm... you wouldn't want the files (the copies inside of the docs/docsite/rst directory) to end up in the sdist, though, as they're generated files.  Does the build error go away because they do get added to the sdist if they're copied?  If so, it's not the correct fix.  Instead we probably want to add an exclude for those files to MANIFEST.in

Reply:
In that case we'll need to ignore them in the sanity test instead of copying them, by adding them here:

https://github.com/ansible/ansible/blob/00bd0b893d5d21de040b53032c466707bacb3b93/test/sanity/code-smell/package-data.py#L38

Reply:
I've got this working.  For posterity, there are several locations that generated files may need to be listed:

* `clean` target of a Makefile so that they're removed when make clean is run
* `MANIFEST.in` they should normally be excluded in here so they do not end up; in the tarball
* `.gitignore` they should be added here so that they aren't checked in to the git repository
* `test/sanity/code-smell/package-data.py` they may need to be added to an ignore list here if we don't want them in the tarball and they aren't ignored by an existing ignore pattern.

Reply:
@samccann what was the reason for not pointing to separate configs?

Reply:
@webknjaz - I'm not sure at this point. I remember I hit a roadblock and had to have @abadger help me out of it. As part of that help, we went to symlinks instead of  the separate -c sphinx option and sphinx tags.

Reply:
@abadger I'd be interested in hearing the details. Over the past year or so I've learned a lot about Sphinx and making Sphinx extensions so maybe I'd be able to help out.

Reply:
I think if we're going to leave the version-switcher in place, we need to keep `index.html` as the name of the default main page.